### PR TITLE
Add artist track page

### PR DESCRIPTION
**Pull request by staniel359**

### CODE REVIEW

File: app/services/vk/track.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def retrieve_track
       encoded_values = link.split('=')[1].split('#')
       first = vk_o(encoded_values[0])
       second = vk_o(encoded_values[1]).split("\v")
-      vk_s(first, second[1].to_i ^ 548_939_802)
+      vk_s(first, second[1].to_i ^ 223_233_444)
```
Comment (Neyaz) [inline app/services/vk/track.rb]: Can you move that number in constant?

File: app/services/vk/track.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@ def retrieve_track
       encoded_values = link.split('=')[1].split('#')
```
Comment (Neyaz) [inline app/services/vk/track.rb]: Maybe better to use pattern matching with named groups
`link.match(/(^.*)(=)(?<encoded_values>.*)(\#)(.*)/i)`

File: app/controllers/artists/tracks_controller.rb [hunk 3]
```diff
@@ -3,20 +3,53 @@ class TracksController < ArtistsController
...
+    def board
+      track
+      title
+      comments
     end
 
   private
 
-    def set_title
+    def title
```
Comment (Neyaz) [inline app/controllers/artists/tracks_controller.rb]: memoization

File: app/services/lastfm/track.rb [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        lastfm_listeners_count: parsed_response['listeners'].to_i,
+        lastfm_plays_count: parsed_response['playcount'].to_i,
+        album: album_data,
+        tags: tags,
+        wiki: wiki
+      }
+    end
+
+    def duration
+      parsed_response['duration'].to_i / 1_000
```
Comment (Neyaz) [inline app/services/lastfm/track.rb]: Constant for number

File: app/services/lastfm/track.rb [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    def album_cover
+      parsed_response.dig(
+        'album', 'image', 3, '#text'
+      ).sub('300x300', '600x600')
+    end
+
+    def tags
+      parsed_response.dig(
+        'toptags', 'tag'
+      ).map { |t| t['name'] }
```
Comment (Neyaz) [inline app/services/lastfm/track.rb]: `.map(&:t['name'])`

File: app/services/lastfm/track/similar_tracks.rb [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+          format: 'json'
+        }
+      end
+
+      def tracks_paginated
+        parsed_response['track'][array_offset, array_limit] || []
+      end
+
+      def array_limit
+        25
```
Comment (Neyaz) [inline app/services/lastfm/track/similar_tracks.rb]: constant

File: app/services/vk/track.rb (hunk 1 above)
Comment (staniel359) [reply app/services/vk/track.rb]: Sure.

File: app/controllers/artists/tracks_controller.rb (hunk 3 above)
Comment (staniel359) [reply app/controllers/artists/tracks_controller.rb]: @Neyaz Do we actually need to memoize it? We are just declaring instance variable for the view.

File: app/services/lastfm/track.rb (hunk 5 above)
Comment (staniel359) [reply app/services/lastfm/track.rb]: @Neyaz It returns `#<Enumerator: ...>`.

File: app/services/vk/track.rb (hunk 2 above)
Comment (staniel359) [reply app/services/vk/track.rb]: @Neyaz Do we need this complex logic? We actually need to get 2 values from a string.

**Pull request merged**